src/libraries/System.Runtime.InteropServices.JavaScript/gen/JSImportGenerator/JSExportCodeGenerator.cs

```diff
@@ -98,6 +98,7 @@ public BlockSyntax GenerateJSExportBody()
             }
 
             tryStatements.AddRange(statements.NotifyForSuccessfulInvoke);
+            tryStatements.AddRange(statements.PinnedMarshal);
```

Comment:
I think `PinnedMarshal` is new stage since the JS gen inception, right ? It's good to have it.

Reply:
Yep! It was added to help generalize some of the support used with the custom marshaller design.